exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -152,13 +143,18 @@ private ProofStatus preCheck() {
    * keys are checked.
    */
   private boolean prefixesIncluded() {
-    return Stream.concat(
-        leaves.stream().map(MapProofEntryLeaf::getDbKey),
-        absentLeaves.stream().map(MapProofAbsentEntryLeaf::getDbKey))
-        .anyMatch(leafEntryKey -> branches.stream()
+    // TODO: entries are sorted, so it's possible to use binary search here
+    List<DbKey> requestedKeys =
+        Stream.concat(
+                leaves.stream().map(l -> DbKey.newLeafKey(l.getKey())),
+                absentLeaves.stream().map(DbKey::newLeafKey))
+            .collect(toList());
+    return requestedKeys
+        .stream()
+        .anyMatch(leafEntryKey -> proof.stream()
             .map(MapProofEntry::getDbKey)
-            .anyMatch(branchEntryKey ->
-                leafEntryKey.commonPrefix(branchEntryKey).equals(branchEntryKey))
+            .anyMatch(proofEntryKey ->
+                leafEntryKey.commonPrefix(proofEntryKey).equals(proofEntryKey))
```

Comment:
I’d suggest adding a `DbKey#isPrefixOf(other: DbKey)` for easier reading, testing and faster implementation (you don't need to create a common prefix to see if one key is a prefix of another):

```java
  public boolean isPrefixOf(DbKey other) {
    return this.keyBits().isPrefixOf(other.keyBits());
  }
```